src/app/directives/image/image.directive.ts

```diff
@@ -63,29 +69,37 @@ export class AuthenticatedImageDirective implements OnChanges {
 
     // On Component Initial Load
     if (changes.src.isFirstChange()) {
-      this.imageRef.nativeElement.onerror = () => {
+      this.imageEl.classList.add("loading-image");
+
+      this.imageEl.onerror = () => {
         // Prevent overriding of 'this'
         this.errorHandler();
       };
+
+      this.imageEl.onload = () => {
+        // Prevent overriding of 'this'
+        this.imageEl.classList.remove("loading-image");
+        this.imageEl.classList.add("loaded-image");
+      };
     }
 
     // Re-enable use of current src
-    if (this.usedUrls.count() > 0) {
-      this.usedUrls = this.usedUrls.delete(this.usedUrls.last());
+    if (this.usedImages.count() > 0) {
+      this.usedImages = this.usedImages.delete(this.usedImages.last());
     }
```

Comment:
random thought; no changes required: this could just be a queue? I.e. pop off the array

Reply:
Off the top of my head, i don't remember why i chose this approach. A queue may be a better approach

Reply:
Had a dig and figured out the problem i originally ran into, however i figured out a bypass which should be a bit cleaner